### PR TITLE
print warning when preconditions are not satisfied + release v0.9.1

### PR DESCRIPTION
**Pull request by AkihiroSuda**
* print warning when unsupported propagation was specified (Suggested in https://github.com/moby/moby/pull/40635#issuecomment-595829139 , CC @thaJeztah @cpuguy83)
* print sysctl warnings
* print warning if running as root

### CODE REVIEW

File: pkg/parent/warn.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/rootless-containers/rootlesskit/pkg/parent/mount"
+)
+
+func warnPropagation(propagation string) {
+	mounts, err := mount.GetMounts()
+	if err != nil {
+		logrus.Warn("Failed to parse mountinfo")
```
Comment (thaJeztah) [inline pkg/parent/warn.go]: are we interested in the error as part of warnings? If so, perhaps `logrus.WithError(err).Warn(...`

File: pkg/parent/warn.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
+package parent
+
+import (
+	"io/ioutil"
+	"strconv"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
```
Comment (thaJeztah) [inline pkg/parent/warn.go]: is this repo using three groups (std, external, local) for imports? If not, (nit) remove this line

File: pkg/parent/warn.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+// warnSysctl verifies /proc/sys/kernel/unprivileged_userns_clone and /proc/sys/user/max_user_namespaces
+func warnSysctl() {
+	uuc, err := ioutil.ReadFile("/proc/sys/kernel/unprivileged_userns_clone")
+	// The file may not exist on all distros
+	if err == nil {
+		i, err := strconv.ParseInt(strings.TrimSpace(string(uuc)), 10, 64)
+		if err == nil && i == 0 {
+			logrus.Warn("/proc/sys/kernel/unprivileged_userns_clone needs to be set to 1.")
```
Comment (thaJeztah) [inline pkg/parent/warn.go]: same question here (WithError?)

File: pkg/parent/mount/mount_linux.go [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+	   (8) separator:  marks the end of the optional fields
+	   (9) filesystem type:  name of filesystem of the form "type[.subtype]"
+	   (10) mount source:  filesystem specific information or "none"
+	   (11) super options:  per super block options*/
+	mountinfoFormat = "%d %d %d:%d %s %s %s %s"
+)
+
+// Parse /proc/self/mountinfo because comparing Dev and ino does not work from
+// bind mounts
+func parseMountTable() ([]*Info, error) {
```
Comment (thaJeztah) [inline pkg/parent/mount/mount_linux.go]: Was this code copied from the moby "mount" pkg?

Should we make that package easier to import, you think? (separate repo or separate module if we're moving to go modules?)

Comment (AkihiroSuda) [reply pkg/parent/mount/mount_linux.go]: It is from runc: https://github.com/rootless-containers/rootlesskit/pull/122/files#diff-eb54b121747b544dfaf8954b8474c615R1

I think this amount of copypasta is fine. Separating the repo is rather likely to cause vendor hell.

File: pkg/parent/warn.go (hunk 2 above)
Comment (AkihiroSuda) [reply pkg/parent/warn.go]: yes, three groups: https://github.com/rootless-containers/rootlesskit/blob/da1dfffa441bad0cfeeb35e7ca6423e5f9ee215d/pkg/parent/parent.go#L4-L26

File: pkg/parent/mount/mount_linux.go (hunk 4 above)
Comment (thaJeztah) [reply pkg/parent/mount/mount_linux.go]: And, looks like runc copied it from moby 😓, so now it's in three locations https://github.com/moby/moby/blob/c4cf72bad325f90ebf021d7cb52c262992a687c7/pkg/mount/mountinfo_linux.go#L120-L130

File: pkg/parent/warn.go (hunk 3 above)
Comment (AkihiroSuda) [reply pkg/parent/warn.go]: done

File: pkg/parent/warn.go (hunk 1 above)
Comment (AkihiroSuda) [reply pkg/parent/warn.go]: done

**Pull request merged**